src/test/java/com/ge/predix/metering/filter/NuregoIT.java

```diff
@@ -16,125 +16,277 @@
-            // if subscription is not created, then create one and copy the subscription ids to the appropriate
-            // subscription id fields above.
-            Map<String, Object> params = new HashMap<>();
-            params.put("plan_id", planId);
-            subscription = Subscription.create(ORGANIZATION_ID, params);
-        }
-        Assert.assertEquals(subscription.getId(), subscriptionId);
-        Double beforeUsedAmount = getEntitlementUsageByFeatureId(featureId, subscriptionId);
-
-        //2-3 second delay doesnt seem to be enough for the nurego server to reflect the udpated count.
-        this.meteringFilter.doFilter(request, response, new MockFilterChain());
-        Thread.sleep(4000);
-        this.meteringFilter.doFilter(request, response, new MockFilterChain());
-        Thread.sleep(4000);
-
-
-
-        Double afterUsedAmount = getEntitlementUsageByFeatureId(featureId, subscriptionId);
-        Assert.assertEquals(afterUsedAmount - beforeUsedAmount, 2.0);
+	private String nuregoToken;
+    
+    private class ComponentUsage{
+    		private int componentCount;
+    		private double amountUsed;
+    		
+    		public ComponentUsage(int componentCount, double amountUsed) {
+    			this.componentCount = componentCount;
+    			this.amountUsed = amountUsed;
+    		}
+    		
+    		public int getCompCount() {
+    			return componentCount;
+    		}
+    		
+    		public double getUsage() {
+    			return amountUsed;
+    		}
+    
     }
+    
+	@BeforeClass
+	public void setUp() throws Exception {
+		nuregoToken = getNuregoAuthToken();
+		// creates an UAA instance and ACS instance
+		UaaInstance uaa = serviceInstanceHelper.createUaaInstance();
+		uaaInstanceId = uaa.getUaaGuid();
+		acsInstanceId = serviceInstanceHelper.createAcsInstance(uaa.getUaaUrl());
 
-    private Double getEntitlementUsageByFeatureId(final String featureId, final String subscriptionId)
-            throws Exception {
-        Entitlement entitlement = getEntitlementByFeatureId(featureId, subscriptionId);
-        if (entitlement == null) {
-            throw new IllegalArgumentException(String.format("Feature '%s' does not exist.", featureId));
-        }
-        return entitlement.getCurrentUsedAmount();
-    }
+	}
 
-    private Entitlement getEntitlementByFeatureId(final String featureId, final String subscriptionId)
-            throws Exception {
-        List<Entitlement> entitlements = Entitlement.retrieve(subscriptionId).getData();
-        for (Entitlement entitlement : entitlements) {
-            if (entitlement.getFeatureId().equals(featureId)) {
-                return entitlement;
-            }
-        }
+    @Test(dataProvider = "requestProviderForOkResponse")
+    public void testNuregoIntegrationForOkResponse(final String featureId, final String planId,
+            final  MockHttpServletRequest request, final ServletResponse response, final boolean acsFlag) throws Exception {
+   
```

Comment:
Since the acs plan id is a constant, you could use that to determine whether it is acs or not instead of adding an extra boolean parameter.